### PR TITLE
Week2 from Rabia

### PR DESCRIPTION
**Pull request by Damla78**

### CODE REVIEW

File: homework-submission/week2/src/Counter/CounterManager.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import React, { Component } from 'react';
+
+export default class CounterManager extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
```
Comment (NouranMahmoud) [inline homework-submission/week2/src/Counter/CounterManager.js]: this.state doesn't have `isChecked`!? how can you use it however it is not defined!?

Comment (Damla78) [reply homework-submission/week2/src/Counter/CounterManager.js]: But I want to take initial value from App.js. How can I pass initial values to checkboxes.

File: homework-submission/week2/src/App.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    "deadline": "2017-09-10",
+    "done": false
+  },
+  {
+    "id": 3,
+    "description": "Eat breakfast",
+    "deadline": "2017-09-09",
+    "done": false
+  }
+]
```
Comment (jalal75) [inline homework-submission/week2/src/App.js]: this todos array should be in you state.

File: homework-submission/week2/src/App.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { isChecked: this.props.done };
+
```
Comment (jalal75) [inline homework-submission/week2/src/App.js]: this.state = { todos : [ ... ] }

File: homework-submission/week2/src/Counter/Controls.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  render{
+  return (
+    <div>
+      <button onClick={() => { this.props.onClick('Start') }}>Start</button>
+      <button onClick={() => { this.props.onClick('Stop') }}>Stop</button>
+      <button onClick={() => { this.props.onClick('Reset') }}>Reset</button>
+    </div >
+  );
+}
+}
```
Comment (jalal75) [inline homework-submission/week2/src/Counter/Controls.js]: this file is not for this home work.

File: homework-submission/week2/src/ToDo/DoneCheckBox.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    "deadline": "2017-09-10",
+    "done": false
+  },
+  {
+    "id": 3,
+    "description": "Eat breakfast",
+    "deadline": "2017-09-09",
+    "done": false
+  }
+]
```
Comment (jalal75) [inline homework-submission/week2/src/ToDo/DoneCheckBox.js]: you are using the state of this component and that is not required, you should pass the state from App component to here as props

File: homework-submission/week2/src/ToDo/DoneCheckBox.js [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+export default class DoneCheckBox extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = { isChecked: this.props.done };
```
Comment (jalal75) [inline homework-submission/week2/src/ToDo/DoneCheckBox.js]: not required

File: homework-submission/week2/src/ToDo/DoneCheckBox.js [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+      isChecked: !this.state.isChecked
+    })
+    /*this.setState(() => (
+      {
+        isChecked: !isChecked,
+      }
+    ));*/
+    console.log('++++++++++' + isChecked);
+    console.log('++++++++++' + !this.state.isChecked);
+  }
```
Comment (jalal75) [inline homework-submission/week2/src/ToDo/DoneCheckBox.js]: this function should also be created in the App and pass it here as props to change the state in the App

**Comment by jalal75** [comment]
Please rewrite the application and try to understand the concept of React.

File: homework-submission/week2/src/App.js (hunk 3 above)
Comment (Damla78) [reply homework-submission/week2/src/App.js]: Hi Jalal. can you explain 'this.state = { todos : [ ... ] }' more? I don't understand how should I write.

Comment (jalal75) [reply homework-submission/week2/src/App.js]: this.state = { todos : [ ... this.props.done] }

**Comment by jalal75** [comment]
It is batter than before but:
you created another state in DoneCheckBox which is not good, because the function handleCheck is effect  DoneCheckBox state and do nothing with your original data in App
so please try to create your state and handleCheck in the App and pass whatever you need as props to children components.

**Comment by jalal75** [comment]
Well done
Nice code, keep doing
I really appreciate your hard work

**Pull request closed without merging**